addons/account/models/account_bank_statement_line.py

```diff
@@ -455,14 +455,16 @@ def _find_or_create_bank_account(self):
         return bank_account.filtered(lambda x: x.company_id in (False, self.company_id))
 
     def _get_default_amls_matching_domain(self):
+        all_reconcilable_accounts = self.env['account.account'].search([("company_id", "child_of", self.company_id.root_id.id), ('reconcile', '=', True)]).ids
```

Comment:
`_check_company_domain`

Reply:
I'd say `child_of` is correct in this case, since we can reconcile lines from any account that allows reconciliation; the account itself doesn't matter, so access to it doesn't matter either.